### PR TITLE
feat(context): Add in foreground and is main thread

### PR DESCRIPTION
**Pull request by krystofwoldrich**
## :scroll: Description
Adding `in_foreground` and `isMain` to keep feature parity.

## :bulb: Motivation and Context
- https://github.com/getsentry/team-mobile/issues/36

## :green_heart: How did you test it?
tests, swift UI example app

## :pencil: Checklist

- [x] I reviewed the submitted code.
- [x] I added tests to verify the changes.
- [x] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled.
- [ ] I updated the docs if needed.
- [ ] Review from the native team if needed.
- [x] No breaking change or entry added to the changelog.
- [x] No breaking change for hybrid SDKs or communicated to hybrid SDKs.

### CODE REVIEW

File: Sources/Sentry/SentryClient.m [hunk 1]
```diff
@@ -540,6 +542,23 @@ - (SentryEvent *_Nullable)prepareEvent:(SentryEvent *)event
...
+                [event.context mutableCopy] ?: [NSMutableDictionary dictionary];
+            if (context[@"app"] == nil || [context[@"app"] isKindOfClass:[NSDictionary self]]) {
+                NSMutableDictionary *app = [(NSDictionary *)context[@"app"] mutableCopy]
+                    ?: [NSMutableDictionary dictionary];
+                context[@"app"] = app;
+
+                app[@"in_foreground"]
+                    = app[@"in_foreground"] == nil ? @(inForeground) : app[@"in_foreground"];
+                event.context = context;
+            }
```
Comment (brustolin) [inline Sources/Sentry/SentryClient.m:559]:
```suggestion
            if (context[@"app"] == nil || [context[@"app"] isKindOfClass:NSDictionary.self] && context[@"app"][@"in_foreground"] == nil ) {
                NSMutableDictionary *app = [(NSDictionary *)context[@"app"] mutableCopy]
                    ?: [NSMutableDictionary dictionary];
                context[@"app"] = app;

                app[@"in_foreground"] = @(inForeground);
                event.context = context;
            }
```

File: Sources/Sentry/SentryCrashReportConverter.m [hunk 2]
```diff
@@ -244,6 +244,8 @@ - (SentryThread *_Nullable)threadAtIndex:(NSInteger)threadIndex
     thread.crashed = threadDictionary[@"crashed"];
     thread.current = threadDictionary[@"current_thread"];
     thread.name = threadDictionary[@"name"];
+    // We don't have access to the MachineContextWrapper but we know first thread is always the main
+    thread.isMain = [NSNumber numberWithBool:[threadDictionary[@"index"] isEqual:@0]];
```
Comment (brustolin) [inline Sources/Sentry/SentryCrashReportConverter.m:248]: No need to deal with the dictionary anymore.
```suggestion
    thread.isMain = [NSNumber numberWithBool:thread.threadId.intValue == 0];
```

**Comment by krystofwoldrich** [comment]
That App with the SDK is a bit larger than the limit in metrics:
411431 should be < 409600

Is it okay considering the changes? Should we adjust the limit?

**Review by brustolin — APPROVED** [review]
LGTM

**Pull request merged**